src/core/api.c

```diff
@@ -1264,6 +1264,22 @@ MsQuicStreamReceiveComplete(
             (uint32_t)QUIC_STATUS_INVALID_STATE);
         goto Exit;
     }
+    
+    if (Connection->WorkerThreadID == CxPlatCurThreadID()) {
+
+        CXPLAT_PASSIVE_CODE();
+
+        BOOLEAN AlreadyInline = Connection->State.InlineApiExecution;
+        if (!AlreadyInline) {
+            Connection->State.InlineApiExecution = TRUE;
+        }
+        QuicStreamReceiveCompleteInline(Stream, BufferLength);
```

Comment:
I think we need to only do this if we in the receive callback, and not just any MsQuic callback.